api/v1/apiserver_types.go

```diff
@@ -87,6 +87,9 @@ type APIServerDeploymentContainer struct {
 	// If used in conjunction with the deprecated ComponentResources, then this value takes precedence.
 	// +optional
 	Resources *v1.ResourceRequirements `json:"resources,omitempty"`
+
+	// +optional
+	Lifecycle *v1.Lifecycle `json:"lifecycle,omitempty" protobuf:"bytes,12,opt,name=lifecycle"`
```

Comment:
I'd like to hear the use case for adding this. (I hadn't heard of Lifecycle before but looking at the K8s API reference I don't understand what its use case would be for the apiserver.)

Reply:
i guess this could be useful if using `preStop` hook to gracefully manage IPAM cleanup when calico-node shut down. But haven't tested this scenario so i'm removing these lines for now. Maybe i can create a new PR later when properly tested it out. Thanks for your comments.

Reply:
removing these lines for now